src/session/secret_chat.rs

```diff
@@ -0,0 +1,152 @@
+use glib::GEnum;
+use gtk::{glib, prelude::*, subclass::prelude::*};
+use tdgrand::enums::{SecretChatState as TdSecretChatState, Update};
+use tdgrand::types::SecretChat as TdSecretChat;
+
+use crate::session::User;
+
+#[derive(Debug, Clone, Copy, PartialEq, GEnum)]
+#[genum(type_name = "SecretChatState")]
+pub enum SecretChatState {
+    Pending,
+    Ready,
+    Closed,
+}
+
+impl Default for SecretChatState {
+    fn default() -> Self {
+        Self::Pending
+    }
+}
+
+impl SecretChatState {
+    pub fn from_td_object(state: &TdSecretChatState) -> Self {
+        match state {
+            TdSecretChatState::Pending => Self::Pending,
+            TdSecretChatState::Ready => Self::Ready,
+            TdSecretChatState::Closed => Self::Closed,
+        }
+    }
+}
+
+mod imp {
+    use super::*;
+    use glib::WeakRef;
+    use once_cell::sync::Lazy;
+    use std::cell::Cell;
+
+    #[derive(Debug, Default)]
+    pub struct SecretChat {
+        pub id: Cell<i32>,
+        pub user: WeakRef<User>,
```

Comment:
Actually, having this as weakref is not really great since we expect to always have an User related to the secret chats.